### PR TITLE
More currencies

### PR DESCRIPTION
**Pull request by Emyrk**

### CODE REVIEW

File: polling/coinmarketcap.go [hunk 1]
```diff
@@ -70,6 +70,9 @@ func (d *CoinMarketCapDataSource) FetchPegPrices() (peg PegAssets, err error) {
 		if !ok {
 			continue
 		}
+		if currency.Symbol != asset && !(asset == "XBT" || asset == "XBC") {
```
Comment (WhoSoup) [inline polling/coinmarketcap.go]: is there any case where this happens? the only time check is true is if CMC changes a token ID to be a different coin. if that happens, then this error could also occur for xbt/xbc. something like:
```
if currency.Symbol != asset && !(asset == "XBT" && currency.Symbol == "BTC") && !(asset == "XBC" && currency.Symbol == "BCH") {
```

File: config/defaultconfig.ini [hunk 2]
```diff
@@ -146,7 +146,6 @@
   AlternativeMe=2
   CoinCap=-1
   PegnetMarketcap=-1
-  Factoshiio=-1
```
[the remark this answers is not in the document]
Comment (Emyrk) [reply config/defaultconfig.ini]: @WhoSoup Alex asked me to start deprecating factoshiio, he won't support it in the future. CMC and Coingecko both have PEG, so we still have 3 sources

File: polling/coinmarketcap.go (hunk 1 above)
Comment (Emyrk) [reply polling/coinmarketcap.go]: This should never happen, but I figured it was a cheap check. I was lazy in checking xbt and xbc, I can include those.

Basically I added this check in another datasouce, then replicated it in a few others as a safety. If we are sure they won't change the token ID I can remove this.

File: modules/grader/v4grader.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
+package grader
+
+var _ BlockGrader = (*V3BlockGrader)(nil)
+
+// V4BlockGrader implements the V2 grading algorithm but requires PEG to have a price.
+type V4BlockGrader struct {
+	V2BlockGrader
```
[the remark this answers is not in the document]
Comment (Emyrk) [reply modules/grader/v4grader.go]: That is a good point. It is a bit awkward. Considering V3 is just V2, I could use v3, but that is an extra indirection which is more difficult to follow imo.

I'll add a comment

File: modules/grader/v4_util.go [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+	if len(o.Assets) != len(opr.V4Assets) {
+		return nil, NewValidateError("invalid assets")
+	}
+	for _, val := range o.Assets {
+		if val == 0 {
+			return nil, NewValidateError("assets must be greater than 0")
+		}
+	}
+
+	if len(o.Winners) != 25 {
```
[the remark this answers is not in the document]
Comment (Emyrk) [reply modules/grader/v4_util.go]: Ah, I see yea. I just kept copying the old code forward. I'll change that here.

File: config/defaultconfig.ini (hunk 2 above)
[the remark this answers is not in the document]
Comment (WhoSoup) [reply config/defaultconfig.ini]: Ah, good to hear that more APIs have added PEG. That alleviates my concern

File: modules/opr/assets.go [hunk 5]
```diff
@@ -73,6 +73,70 @@ var V2Assets = []string{
...
+	"ZAR",
+	"TRY",
+	"EOS",
+	"LINK",
+	"ATOM",
+	"BAT",
+	"XTZ",
+
+	// Reference Tokens
+	"pUSD",
```
Comment (WhoSoup) [inline modules/opr/assets.go]: -"pUSD"

File: common/assets.go [hunk 6]
```diff
@@ -92,14 +92,39 @@ var (
...
+	AssetsV1  = MergeLists(PEGAsset, CurrencyAssets, CommodityAssets, CryptoAssets)
 	// This is with the PNT instead of PEG. Should never be used unless absolutely necessary.
 	//
 	// Deprecated: Was used for version 1 before PNT -> PEG
 	AssetsV1WithPNT = MergeLists([]string{"PNT"}, SubtractFromSet(AssetsV1, "PEG"))
 	// Version One, subtract 2 assets
 	AssetsV2 = SubtractFromSet(AssetsV1, "XPD", "XPT")
+
+	// Additional assets to V2 set
+	AssetsV4 = MergeLists(AssetsV2, V4CurrencyAdditions, V4CryptoAdditions, V4ReferenceAdditions)
```
Comment (WhoSoup) [inline common/assets.go]: -V4ReferenceAdditions

**Pull request merged**